src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseServiceIntegrationTest.java

```diff
@@ -156,6 +157,34 @@ void testInstructorGetsResultsFromOwningCoursesNotEmpty() throws Exception {
         assertThat(result.getResultsOnPage()).hasSize(1);
     }
 
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    void testInstructorSearchTermMatchesId() throws Exception {
+        database.resetDatabase();
+        database.addUsers(1, 1, 0, 1);
+        testSearchTermMatchesId();
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    void testAdminSearchTermMatchesId() throws Exception {
+        database.resetDatabase();
+        database.addUsers(1, 1, 0, 1);
+        testSearchTermMatchesId();
+    }
```

Comment:
I agree with @Dominik-Weinzierl here about code duplication. You can easily refactor this code snippet by extracting the method bodies into a new function. 

However, my suggestion would be to unite these 2 tests into a single parametrized test and see if you can pass the active user and user role as parameters. You can use `database.changeUser("tutor1");` for instance, to change the active user while a test is running.

Reply:
I tried this already. When I do this I get the following error when I run the test with switch user to "instructor1"
```
Authentication object cannot be null; nested exception is java.lang.IllegalArgumentException: Authentication object cannot be null
org.springframework.dao.InvalidDataAccessApiUsageException: Authentication object cannot be null; nested exception is java.lang.IllegalArgumentException: Authentication object cannot be null
```

Reply:
Done in d4f280d9e20ba7a46f3fa39b0ef87c48de2cce5c

Reply:
@canberkanar I disagree in this case. Test with separate roles -> Separate Tests. Tests should only test one specific scenario